pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -248,6 +261,76 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, kc *v1beta1.KafkaChannel
 	return newReconciledNormal(kc.Namespace, kc.Name)
 }
 
+func (r *Reconciler) setupSubscriptionStatusWatchers(ctx context.Context, channel *v1beta1.KafkaChannel) error {
+	var err error
+	if channel.Spec.SubscribableSpec.Subscribers != nil {
+		for _, s := range channel.Spec.SubscribableSpec.Subscribers {
+			groupID := fmt.Sprintf("kafka.%s.%s.%s", channel.Namespace, channel.Name, string(s.UID))
+			err = r.kafkaWatcher.WatchCosumerGroup(groupID, func(event CGEvent) {
+				var err error
+				switch event {
+				case CGObserved:
+					err = r.markSubscriptionReadiness(ctx, channel, &s, true)
+
+				case CGDeleted:
+					err = r.markSubscriptionReadiness(ctx, channel, &s, false)
+				}
+				if err != nil {
+					logging.FromContext(ctx).Errorw("error updating subscription readiness", zap.Error(err))
+				}
+			})
+		}
+
+	}
+	return err
+}
+
+func (r *Reconciler) markSubscriptionReadiness(ctx context.Context, ch *v1beta1.KafkaChannel, sub *v1.SubscriberSpec, ready bool) error {
+	after := ch.DeepCopy()
+	found := false
+	after.Status.Subscribers = make([]v1.SubscriberStatus, 0)
+
+	for _, s := range ch.Status.Subscribers {
+		if s.UID == sub.UID {
+			found = true
+			if !ready {
+				// we need to make the subscription not ready, which means
+				// skip adding it to Status.Subscribers
+				continue
+			}
+		}
+		after.Status.Subscribers = append(after.Status.Subscribers, s)
+	}
+	if !found && ready {
```

Comment:
what happens when a subscription has been deleted? This function will be triggered when the consumer group is purged from Kafka. Adding it again to status.subscriber feels wrong.

Reply:
I refactored that whole part so might be good to check the code again